nitorch/nn/training/_training.py

```diff
@@ -563,15 +574,20 @@ def train(self):
                 self._eval(self.epoch)
                 self._save(self.epoch)
                 for self.epoch in range(self.epoch+1, self.nb_epoch+1):
-                    self._train(self.epoch)
-                    self._eval(self.epoch)
+                    _ = self._train(self.epoch)
+                    val_loss = self._eval(self.epoch)
                     self._save(self.epoch)
-
+                    # incorporate learning rate scheduler
+                    if self.scheduler is not None:
+                        if isinstance(self.scheduler, ReduceLROnPlateau):
```

Comment:
I wonder if it is possible to inspect the scheduler (using the built in  `inspect` package) to know if a loss must be provided or not. That would be more generic.

Reply:
Looking at the implemented scheds in torch.optim.lr_scheduler, ReduceLROnPlateau is the only one that takes an input to its step call. So maybe it is okay to keep it like this for now?